ReactWindows/ReactNative.ChakraBridge/JsrtJavaScriptExecutor.cpp

```diff
@@ -0,0 +1,190 @@
+#include "pch.h"
+#include "JsrtJavaScriptExecutor.h"
+
+using namespace ReactNative::ChakraBridge;
+
+int ChakraJavaScriptExecutor::InitializeHost()
+{
+	return this->host.Init();
+}
+
+int ChakraJavaScriptExecutor::DisposeHost()
+{
+	return this->host.Destroy();
+}
+
+int ChakraJavaScriptExecutor::SetGlobalVariable(String^ variableName, String^ stringifiedText)
+{
+	JsValueRef valueStringified;
+	IfFailRet(JsPointerToString(stringifiedText->Data(), stringifiedText->Length(), &valueStringified));
+
+	JsValueRef valueJson;
+	IfFailRet(this->host.JsonParse(valueStringified, &valueJson));
+	IfFailRet(this->host.SetGlobalVariable(variableName->Data(), valueJson));
+
+	return JsNoError;
+}
+
+ChakraStringResult ChakraJavaScriptExecutor::GetGlobalVariable(String^ variableName)
+{
+	JsValueRef globalVariable;
+	IfFailRetNullPtr(this->host.GetGlobalVariable(variableName->Data(), &globalVariable));
+
+	JsValueRef globalVariableJson;
+	IfFailRetNullPtr(this->host.JsonStringify(globalVariable, &globalVariableJson));
+
+	const wchar_t* szBuf;
+	size_t bufLen;
+	IfFailRetNullPtr(JsStringToPointer(globalVariableJson, &szBuf, &bufLen));
+
+	ChakraStringResult finalResult = { JsNoError, ref new String(szBuf, bufLen) };
+	return finalResult;
+}
+
+ChakraStringResult ChakraJavaScriptExecutor::RunScript(String^ source, String^ sourceUri)
+{
+	JsValueRef result;
+	IfFailRetNullPtr(this->host.RunScript(source->Data(), sourceUri->Data(), &result));
+
+	JsValueRef resultJson;
+	IfFailRetNullPtr(this->host.JsonStringify(result, &resultJson));
+
+	const wchar_t* szBuf;
+	size_t bufLen;
+	IfFailRetNullPtr(JsStringToPointer(resultJson, &szBuf, &bufLen));
+
+	ChakraStringResult finalResult = { JsNoError, ref new String(szBuf, bufLen) };
+	return finalResult;
+}
+
+IAsyncOperation<ChakraStringResult>^ ChakraJavaScriptExecutor::RunScriptFromFile(String^ sourceUri)
+{
+	Uri^ fileUri = ref new Uri(sourceUri);
+	return create_async([this, fileUri, sourceUri]
+	{
+		return create_task(StorageFile::GetFileFromApplicationUriAsync(fileUri))
+			.then([this, sourceUri](StorageFile^ storageFile)
+			{
+				return create_task(FileIO::ReadTextAsync(storageFile));
```

Comment:
>ReadTextAsync [](start = 31, length = 13)

Quick question... if this is returning a Platform::String^ anyway, isn't it still ending up on the large object heap?